src/js/hca/helpers.jsx

```diff
@@ -170,50 +170,69 @@ export const expensesGreaterThanIncomeWarning = (
   </div>
 );
 
-export function expensesLessThanIncome(formData) {
-  const {
-    veteranGrossIncome = 0,
-    veteranNetIncome = 0,
-    veteranOtherIncome = 0,
-    dependents = []
-  } = formData;
-
-  const {
-    spouseGrossIncome = 0,
-    spouseNetIncome = 0,
-    spouseOtherIncome = 0
-  } = formData['view:spouseIncome'] || {};
-
-  const vetSpouseIncome =
-    veteranGrossIncome +
-    veteranNetIncome +
-    veteranOtherIncome +
-    spouseGrossIncome +
-    spouseNetIncome +
-    spouseOtherIncome;
-
-  const income = dependents.reduce((sum, dependent) => {
+export function expensesLessThanIncome(fieldShownUnder) {
+  const fields = ['deductibleMedicalExpenses', 'deductibleFuneralExpenses', 'deductibleEducationExpenses'];
+  return (formData) => {
     const {
-      grossIncome = 0,
-      netIncome = 0,
-      otherIncome = 0,
-    } = dependent;
-
-    return grossIncome + netIncome + otherIncome + sum;
-  }, vetSpouseIncome);
-
-  const {
-    deductibleMedicalExpenses = 0,
-    deductibleFuneralExpenses = 0,
-    deductibleEducationExpenses = 0,
-  } = formData;
-
-  const expenses =
-    deductibleMedicalExpenses +
-    deductibleEducationExpenses +
-    deductibleFuneralExpenses;
-
-  return income > expenses;
+      veteranGrossIncome = 0,
+      veteranNetIncome = 0,
+      veteranOtherIncome = 0,
+      dependents = []
+    } = formData;
+
+    const {
+      spouseGrossIncome = 0,
+      spouseNetIncome = 0,
+      spouseOtherIncome = 0
+    } = formData['view:spouseIncome'] || {};
+
+    const vetSpouseIncome =
+      veteranGrossIncome +
+      veteranNetIncome +
+      veteranOtherIncome +
+      spouseGrossIncome +
+      spouseNetIncome +
+      spouseOtherIncome;
+
+    const income = dependents.reduce((sum, dependent) => {
+      const {
+        grossIncome = 0,
+        netIncome = 0,
+        otherIncome = 0,
+      } = dependent;
+
+      return grossIncome + netIncome + otherIncome + sum;
+    }, vetSpouseIncome);
+
+    const {
+      deductibleMedicalExpenses = 0,
+      deductibleFuneralExpenses = 0,
+      deductibleEducationExpenses = 0,
+    } = formData;
+
+    const expenses =
+      deductibleMedicalExpenses +
+      deductibleEducationExpenses +
+      deductibleFuneralExpenses;
+
+    const hideBasedOnValues = income > expenses;
+
+    // If we're not going to hide based on values entered,
+    // then we need to make sure the current field is the last non-empty field
+    if (!hideBasedOnValues) {
+      const nonEmptyFields = fields.filter(field => formData[field]);
+      // If all fields are filled in, put it under the last one
+      if (!nonEmptyFields.length && fieldShownUnder === fields[fields.length - 1]) {
```

Comment:
According to the comment on line 224, should this be `nonEmptyFields.length === 3` (i.e. the expected total number have been filled in)? Or would it make sense to change the filter condition on line 223 to be `!formData[field]` so that empty fields are captured and `!emptyFields.length` can be used here instead? Is it necessary to have this clause? Could just the other two suffice?

Reply:
Yep, looks like I just need the one.